eureka2-client/src/main/java/com/netflix/eureka2/client/EurekaClientBuilder.java

```diff
@@ -101,13 +101,13 @@ public EurekaClient build() {
             transportConfig = new BasicEurekaTransportConfig.Builder().build();
         }
 
-        InterestHandler interestHandler = doDiscovery ? buildInterestHandler() : null;
-        RegistrationHandler registrationHandler = doRegistration ? buildRegistrationHandler() : null;
+        EurekaInterestClient interestClient = doDiscovery ? buildInterestClient() : null;
+        EurekaRegistrationClient registrationClient = doRegistration ? buildRegistrationClient() : null;
 
-        return new EurekaClientImpl(interestHandler, registrationHandler);
+        return new EurekaClientImpl(interestClient, registrationClient);
     }
 
-    private InterestHandler buildInterestHandler() {
+    private EurekaInterestClient buildInterestClient() {
```

Comment:
I think we should split the builder in the same way EurekaClient was split. Now configuration properties specific to registration/interest subscription are mixed together. They are not many, but if we add more it will be less transparent to users which one are needed/important in a particular case.